util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/stub/mapper/StubMapperUtils.java

```diff
@@ -20,4 +31,28 @@ public static DocumentReferenceProvider referenceProvider(String identifier) {
 	public static DocumentReferenceProvider referenceProvider(String identifier, String routingKey) {
 		return new StubDocumentReferenceProvider( identifier, routingKey );
 	}
+
+	public static <R, O> void expectLoad(ObjectLoader<R, O> objectLoaderMock,
+			Consumer<LoadingDefinitionContext<R, O>> loadingDefinition) {
+		LoadingDefinitionContext<R, O> context = new LoadingDefinitionContext<>();
+		loadingDefinition.accept( context );
+
+		EasyMock.expect( objectLoaderMock.load(
+				EasyMockUtils.collectionAnyOrderMatcher( new ArrayList<>( context.loadingMap.keySet() ) )
```

Comment:
I think there is no reason here to match the exact parameter, maybe you could use just `anyObject(List.class)`. 
So that we could remove the inner helper method.

Reply:
The idea was to check the references are correctly transformed by the backend. If they are not, the load call will fail and tell us the parameters were not the expected ones.

I suppose something would ultimately fail too (because of null loaded objects) if the wrong parameters were passed to these methods, but the error message would probably be less clear?

Reply:
You're right, in this case it makes sense to check that the backend will produce the expected references. I guess the same.